### PR TITLE
[wasm][debugger] Fix chinese character in project name

### PR DESCRIPTION
**Pull request by thaystg**
The test case does not simulate the real case from blazor, but I added it anyway.

From blaze we receive from the loaded_files variable file names like this:
http://localhost:5168/_framework/%E3%84%A8U1U2U3.Shared.dll
Then we need to convert to: http://localhost:5168/_framework/ㄨU1U2U3.Shared.dll to find it in our assemblies by name.

In the debugger-tests in the mono-config.json, the file name is like this, we add and search with this name, so we don't need the UnescapeDataString.
```
{
      "behavior": "assembly",
      "name": "debugger-test-chinese-char-in-path-\u3128.dll"
}
```

### CODE REVIEW

File: src/mono/wasm/debugger/BrowserDebugProxy/DebugStore.cs [hunk 1]
```diff
@@ -1469,11 +1469,12 @@ public async IAsyncEnumerable<SourceFile> Load(SessionId id, string[] loaded_fil
                         continue;
                     try
                     {
+                        string decodedFileName = Uri.UnescapeDataString(file_name);
```
Comment (radical) [inline src/mono/wasm/debugger/BrowserDebugProxy/DebugStore.cs:1472]: nit: `unescapedFileName`

File: src/mono/wasm/debugger/tests/debugger-test-chinese-char-in-path-ㄨ/debugger-test-chinese-char-in-path-ㄨ.csproj [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <PublishWindowsPdb>false</PublishWindowsPdb>
```
Comment (radical) [inline src/mono/wasm/debugger/tests/debugger-test-chinese-char-in-path-ㄨ/debugger-test-chinese-char-in-path-ㄨ.csproj:3]: Why is this needed?

**Comment by radical** [comment]
Can we test this case against other ways of loading assemblies?
- lazy loading
- hot reload

File: src/mono/wasm/debugger/tests/debugger-test-chinese-char-in-path-ㄨ/debugger-test-chinese-char-in-path-ㄨ.csproj (hunk 2 above)
Comment (thaystg) [reply src/mono/wasm/debugger/tests/debugger-test-chinese-char-in-path-ㄨ/debugger-test-chinese-char-in-path-ㄨ.csproj:3]: Without this we get an error on CI:

`The command ""C:\Users\itomkowicz\source\repos\runtime-fork\.packages\microsoft.diasymreader.pdb2pdb\1.1.0-beta2-19575-01\tools\Pdb2Pdb.exe" "C:\Users\itomkowicz\source\repos\runtime-fork\artifacts\bin\debugger-test\Debug\wasm\debugger-test-special-char-in-path.dll" /out "C:\Users\itomkowicz\source\repos\runtime-fork\artifacts\SymStore\Debug\debugger-test-special-char-in-path\net6.0\x64\debugger-test-special-char-in-path.pdb" /srcsvrvar SRC_INDEX=public" exited with code -1.`

File: src/mono/wasm/debugger/tests/ApplyUpdateReferencedAssemblyChineseCharInPathㄨ/MethodBody1_v2.cs [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    public class MethodBody4 {
+        public static void StaticMethod4 () {
+        }
+    }
+
+
+
+
+
+
```
Comment (radical) [inline src/mono/wasm/debugger/tests/ApplyUpdateReferencedAssemblyChineseCharInPathㄨ/MethodBody1_v2.cs:46]: Are these deliberate?

Comment (thaystg) [reply src/mono/wasm/debugger/tests/ApplyUpdateReferencedAssemblyChineseCharInPathㄨ/MethodBody1_v2.cs:46]: yes, I don't want to move the line of definition of MethodBody5.

File: src/mono/wasm/debugger/tests/debugger-test-chinese-char-in-path-ㄨ/debugger-test-chinese-char-in-path-ㄨ.csproj (hunk 2 above)
Comment (radical) [reply src/mono/wasm/debugger/tests/debugger-test-chinese-char-in-path-ㄨ/debugger-test-chinese-char-in-path-ㄨ.csproj:3]: Any idea what would cause this? And details before that line?

Comment (thaystg) [reply src/mono/wasm/debugger/tests/debugger-test-chinese-char-in-path-ㄨ/debugger-test-chinese-char-in-path-ㄨ.csproj:3]: Pdb2Pdb.exe does not support special characteres in path, as far as I remember.

Comment (radical) [reply src/mono/wasm/debugger/tests/debugger-test-chinese-char-in-path-ㄨ/debugger-test-chinese-char-in-path-ㄨ.csproj:3]: Ok, we can add a comment for that here. And if there is any supporting link with more info, then add that.

Comment (radical) [inline src/mono/wasm/debugger/tests/debugger-test-chinese-char-in-path-ㄨ/debugger-test-chinese-char-in-path-ㄨ.csproj:3]:
```suggestion
    <!-- pdb2pdb.exe does not seem to support special characters in the path, and exits with -1 -->
    <PublishWindowsPdb>false</PublishWindowsPdb>
```

**Review by radical — APPROVED** [review]
